DigitalLearningSolutions.Web.Tests/Attributes/WholeNumberWithinRangeAttributeTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿namespace DigitalLearningSolutions.Web.Tests.Attributes
+{
+    using DigitalLearningSolutions.Web.Attributes;
+    using FluentAssertions;
+    using NUnit.Framework;
+
+    public class WholeNumberWithinRangeAttributeTests
+    {
+        [Test]
+        public void Accepts_number_within_range()

```

Comment:
You could reduce these to a single test using `[TestCaseSource]` like in LinkedFieldHelperTests.cs.

It's nice to have the names of the tests to tell us what each case is for. You could keep that by passing a failure message in with the test data and giving it to NUnit for when the test fails. I believe FluentAssertions lets you do that quite easily. [This article](https://fluentassertions.com/introduction#subject-identification) should send you in the right direction, though may not have everything you need.

Reply:
I've used [TestCaseSource] to refactor the tests and I'm passing in a custom test name with each one :)